### PR TITLE
updated all packages to use @solana/web3.js as a peer dependency

### PR DESCRIPTION
**Pull request by biw**
Going off the same issues as outlined in #340, this PR updates all packages to use `@solana/web3.js` as a peer dependency. 

I don't see anything breaking with this change as the readme directs users to install `@solana/web3.js` as a dependency.

### CODE REVIEW

**Comment by jordansexton** [comment]
So this looks good, but I think these packages now need web3.js as a dev dependency so it gets installed when building each one from source.

**Comment by biw** [comment]
Good catch, updated! Thanks @jordansexton

**Review by alirezaayande — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -2300,6 +2300,20 @@
     prop-types "^15.7.2"
     react-is "^17.0.2"
 
+"@mui/base@5.0.0-alpha.77":
```
Comment (jordansexton) [inline yarn.lock:2303]: Hmm, can you figure out why this is getting included here? It seems like this file shouldn't need to change at all if we just moved some deps into peer and dev deps, right?

**Comment by jordansexton** [comment]
Looks great, I made the order of peer/deps/devDeps consistent. I'm wondering why the yarn lockfile changes when running `yarn install` in this branch though.

**Comment by jordansexton** [comment]
Nevermind the yarn.lock issue, it looks like it's just a result of some nested dependency update. Thanks for this!

**Pull request merged**